src/saleor_app/deps.py

```diff
@@ -121,6 +121,16 @@ async def verify_webhook_signature(
         )
 
 
+async def verify_issuing_principal_type(request: Request):
```

Comment:
This should or could be a class-based dependency:
```python
from saleor_app.schemas.webhook import Webhook


class IgnoredIssuingPrincipalChecker:
    def __init__(self, principal_id: str, raise_exception: bool = True):
        self.principal_id = principal_id
        self.raise_exception = raise_exception
    async def __call__(self, request: Request):
        data = Webhook.parse_raw(await request.body())
        if meta := getattr(data, "meta", None):
            # do the meta check 
            if self.raise_exception:
                raise IgnoredPrincipal()
```
then in an app (not globally in the framework)

```python
@app.exception_handler(IgnoredPrincipal)
async def ignored_principal_exception_handler(request: Request, exc: IgnoredPrincipal):
    return JSONResponse(
        status_code=200,
        content={
            "message": f"Principal Ignored: {exc.saleor_event_type}.",
            "success": True,
        },
    )

is_principal_ignored = IgnoredIssuingPrincipalChecker("my-saleor-app", raise_exception=true)

async def product_created(
    payload: List[Webhook], saleor_domain=Depends(saleor_domain_header), _is_principal_ignored=Depends(is_principal_ignored)
):
    print("Product created!")
    print(payload)
```

(code written on "a knee" might not work)

Reply:
Done